packages/blob/src/helpers.ts

```diff
@@ -32,12 +35,65 @@ export class BlobAccessError extends Error {
   }
 }
 
+export class BlobNotFoundError extends Error {
+  constructor() {
+    super('Vercel Blob: This store does not exist');
+  }
+}
+
+export class BlobSuspendedError extends Error {
+  constructor() {
+    super('Vercel Blob: This store has been suspended');
+  }
+}
+
 export class BlobUnknownError extends Error {
   constructor() {
     super('Vercel Blob: Unknown error, please visit https://vercel.com/help');
   }
 }
 
+type BlobApiErrorCodes =
+  | 'suspended_store'
+  | 'forbidden'
+  | 'not_found'
+  | 'bad_request';
+
+type BlobApiError = { error?: { code: BlobApiErrorCodes } } | undefined;
+
+export async function validateBlobApiResponse(
+  response: Response
+): Promise<unknown> {
+  if (response.status !== 200) {
+    if (response.status < 500) {
+      let data: unknown;
+      try {
+        data = await response.json();
+      } catch {
+        // ignore
```

Comment:
Curious: why ignoring instead of immediately throwing an unknown error here? if we can't response.json then we know it's an unknown error

Reply:
From what I can see, the 404 JSON response is not empty in our API, it always contains JSON data. If it's empty then that would be an unknown error.

Reply:
ok, I updated the test so it matches the worker response (we may need to review this response though it's irrelevant for now as we don't read it)